### PR TITLE
Add model classes for PermissionOffer

### LINKED ISSUE #1046: Client permission API

### PR DESCRIPTION
**Pull request by nirinchev**
Resolves #1046

### CODE REVIEW

File: Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferPCL.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+namespace Realms.Sync
+{
+    /// <summary>
+    /// Objects of this class are used to offer permissions to owned Realms.
+    /// They are created exclusively by the client and are processed by the server
+    /// as indicated by the status fields.
+    /// </summary>
+    /// <remarks>
+    /// When offering permissions, you should create the offer and add it to the <see cref="User"/>'s Management Realm.
```
Comment (bmunkholm) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferPCL.cs]: below lines should not be longer than this line.

File: Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferPCL.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        /// </summary>
+        public bool MayWrite { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the receiver of this offer will be able to manage access rights for others.
+        /// </summary>
+        public bool MayManage { get; }
+
+        /// <summary>
+        /// Gets the expiration date of the offer.
```
Comment (bmunkholm) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferPCL.cs]: date -> date and time

File: Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferResponsePCL.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+using System;
+
+namespace Realms.Sync
+{
+    /// <summary>
+    /// Objects of this class are used to accept a <see cref="PermissionOffer"/> using a provided <see cref="Token"/>.
+    /// </summary>
+    /// <remarks>
+    /// Create an instance of <see cref="PermissionOfferResponse"/> using the provided <see cref="PermissionOffer.Token"/> and add it
```
Comment (bmunkholm) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferResponsePCL.cs]: Again here. Shorter lines (lines should be readable without scrolling in github :-)

File: Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferResponsePCL.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            Token = token;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PermissionOfferResponse"/> class.
+        /// </summary>
+        /// <remarks>
+        /// This constructor is used by Realm to create a new instance from C++. You should use the constructor that accepts parameters instead.
+        /// </remarks>
+        public PermissionOfferResponse()
```
Comment (nirinchev) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionOfferResponsePCL.cs]: Yes, I'm making it private, haven't pushed that yet though :)

**Pull request merged**